features/include/pcl/features/normal_3d.h

```diff
@@ -390,6 +390,67 @@ namespace pcl
         }
       }
       
+      /** \brief Compute the normalized 3x3 covariance matrix and the centroid of a given set of points in a single loop.
+      * Normalized means that every entry has been divided by the number of valid entries in the point cloud.
+      * For small number of points, or if you want explicitly the sample-variance, scale the covariance matrix
+      * with n / (n-1), where n is the number of points used to calculate the covariance matrix and is returned by this function.
+      * Prior to the covariance matrix calculation the center point of the neighborhood is subtracted from all points.
+      * This is done to avoid numerical issues due to loss of significance.
+      * \note This method does not check whether the cloud is dense because it expects finite neighborhoods.
+      * \param[in] cloud the input point cloud
+      * \param[in] indices subset of points given by their indices
+      * \param[out] covariance_matrix the resultant 3x3 covariance matrix
+      * \param[out] centroid the centroid of the set of points in the cloud
+      * \return size of the input cloud.
+      * \ingroup common
+      */
+      inline unsigned int
+      computeMeanAndCovarianceMatrixDemeaned (const pcl::PointCloud<PointInT> &cloud,
+                                              const std::vector<int> &indices,
+                                              Eigen::Matrix<float, 3, 3> &covariance_matrix,
+                                              Eigen::Matrix<float, 4, 1> &centroid)
+      {
+      // create the buffer on the stack which is much faster than using cloud[indices[i]] and centroid as a buffer
+      Eigen::Matrix<float, 1, 9, Eigen::RowMajor> accu = Eigen::Matrix<float, 1, 9, Eigen::RowMajor>::Zero ();
+      size_t point_count;
+      PointInT center = cloud[indices[0]];
```

Comment:
Take a const reference the point.

Reply:
The case if indices is an empty vector, e.g. no neighbors, must be considered.

Reply:
Add that extra logic then. My original inline comment is still applicable.